File: csharp/DiagramPrinter/PagesBuilder.cs
namespace DiagramPrinter;

public class PagesBuilder
{
    public DiagramReportPage ProcessPage(DiagramPage page, FlowchartReportItems data)
    {
        page.AddDataFrom(data);
        return new DiagramReportPage(page);
    }

    public bool Apply(DiagramPagesReport report, List<DiagramReportPage> reportPages)
    {
        if (report.IsComplete())
        {
            return false;
        }
        foreach (var page in reportPages)
        {
            report.AddPage(page);
        }

        return true;
    }
}

File: java/src/main/java/sammancoaching/PagesBuilder.java

package sammancoaching;

import java.util.List;

public class PagesBuilder {
    public DiagramReportPage processPage(DiagramPage page, FlowchartReportItems data) {
        page.addDataFrom(data);
        return new DiagramReportPage(page);
    }

    public boolean apply(DiagramPagesReport report, List<DiagramReportPage> reportPages) {
        if (report.isComplete()) {
            return false;
        }

        for (DiagramReportPage page : reportPages) {
            report.addPage(page);
        }

        return true;
    }
}
